scripts/fn_paradrop.sqf

```diff
@@ -0,0 +1,213 @@
+        (fullCrew [_vehicle,"cargo"]) isEqualTo []
+    }, {
+        params ["_vehicle", "_origHeight"];
+        _vehicle flyInHeight _origHeight;
+    }, [_vehicle, _origHeight]] call CBA_fnc_waitUntilAndExecute;
+};
+
+// next process waypoints if waypoints desired. 
+// if waypoint is desired, then script will be run on setWaypointStatements
+// if not then script will be run in this function.
+
+// Use addWayPoint for the group piloting the vehicle to add the drop waypoints
+private _dropRunOrigin = getPosATL _vehicle; // calling position from leader will call last position on the ground
+
+// Obtain a random location within drop target
+// Between current position and drop target, find angle
+private _dropTarget = [_loc, true] call CBA_fnc_randPosArea;
+
+if (_dropTarget isEqualTo []) exitWith {};
+
+    diag_log [_pilotGroup];
+private _fnc_dzLocs = {
+    params ["_dropTarget","_dropRunOrigin","_dropLength"];
+    // getDir gives the azimuth, which differs in convention from trig
+    // in two ways: (1) azimuth has zero origin at North and
+    // (2) advances clockwise rather than counterclockwise
+    // To convert, use a conditional function:    
+    private _azimuth = _dropRunOrigin getDir _dropTarget;
+    private _theta = switch (floor _azimuth/90) do {
+        case 0: {90-_azimuth;};
+        default {450-_azimuth;};
+    };
+    // For a known drop point and known radius with trig angle _theta,
+    // and radius r given by _dropLength/2,
+    // then coordinates in the xy-plane relative to _dropTarget are:
+    // +/- r*cos _theta for x (Northing) and +/- r*sin _theta for x (Easting)
+
+    private _yDiff = (sin _theta) * _dropLength/2;
+    private _xDiff = (cos _theta) * _dropLength/2;
+    // set all coordinates based on DZ points in xy-plane; z coord does not matter
+    // double lengths on second DZ coordinate to ensure straight line flight
+    [[(_dropTarget select 0) + _xDiff,(_dropTarget select 1) + _yDiff,_dropHeight],
+        [(_dropTarget select 0) - _xDiff,(_dropTarget select 1) - _yDiff,_dropHeight]];
+};
+
+private _dropZoneLocs = [_dropTarget, _dropRunOrigin, _dropLength] call _fnc_dzLocs;
+private _minDistance = _dropRunOrigin distance (_dropZoneLocs select 1);
+
+// Handling for when calling waypoint is within distance _dropLength
+if (_minDistance < _dropLength) then {
+    _dropLength = _minDistance - 100; //arbitrary distance to allow for a helicopter turn
+    if (_dropLength < 0) exitWith {ERROR_1("Waypoint too close to DZ for drop!")};
+    _dropZoneLocs = [_dropTarget,_dropRunOrigin, _dropLength] call _fnc_dzLocs;
+}; 
+
+private _dp = _pilotGroup addWaypoint [_dropZoneLocs select 1,-1,currentWaypoint _pilotGroup + 1,""];
+if (!_disableWP2) then {
+    private _de = _pilotGroup addWaypoint [_dropZoneLocs select 0,-1,currentWaypoint _pilotGroup + 2,""];
+};
+
+// Add the paradrop operation. To store variables, adds info to vehicle characteristics hash that can be referenced globally
```

Comment:
```suggestion
// Add the paradrop operation. To store variables, adds info to vehicle name space that can be referenced later
```
It doesn't set those variables globally, just in the vehicle namespace (and only on the local client).